### PR TITLE
giphy and more

### PR DESCRIPTION
**Pull request by aulneau**

### CODE REVIEW

File: src/common/addresses.ts [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+  }
+  return c32address(_version, hash160);
+};
+
+/**
+ * Convert address
+ *
+ * Converts a STACKS address to a given network mode (testnet/mainnet)
+ */
+
```
Comment (fbwoolf) [inline src/common/addresses.ts:81]: Extra newline here.

File: src/common/hooks/use-boolean.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import { useAtom } from 'jotai';
+import { booleanAtom } from '@store/ui';
+
+export function useToggle(key: string) {
```
Comment (fbwoolf) [inline src/common/hooks/use-boolean.ts:4]: Seems confusing the hook is called `useToggle` if the file is `use-boolean`?

File: src/common/hooks/use-publish-hey.ts [hunk 3]
```diff
@@ -41,7 +44,10 @@ export function useHandlePublishContent() {
...
         contractName,
         functionName: MESSAGE_FUNCTION,
-        functionArgs: [stringUtf8CV(content)],
+        functionArgs: [
+          stringUtf8CV(content),
+          attachmentUri !== '' ? someCV(stringUtf8CV(attachmentUri)) : noneCV(),
+        ],
         onFinish: () => {
           _onFinish();
           onFinish();
```
Comment (fbwoolf) [inline src/common/hooks/use-publish-hey.ts:53]: I don't get this? What is it doing here `onFinish`?

File: src/components/user-area.tsx [hunk 4]
```diff
@@ -63,9 +63,10 @@ const AccountNameComponent = memo(() => {
   const { user } = useUser();
   // Temporarily getting names from mainnet
   const address = useCurrentMainnetAddress();
+  const testnetAddress = useCurrentAddress();
   const names = useAccountNames(address);
   const name = names?.[0];
-  return <Text mb="tight">{name || user?.username || truncateMiddle(address)}</Text>;
+  return <Text mb="tight">{name || user?.username || truncateMiddle(testnetAddress)}</Text>;
```
Comment (fbwoolf) [inline src/components/user-area.tsx:69]: Good catch, thx!

File: src/store/names.ts [hunk 5]
```diff
@@ -1,30 +1,18 @@
...
 const makeKey = (networkUrl: string, address: string): string => {
   return `${networkUrl}__${address}`;
 };
+
 interface Param {
   networkUrl: string;
   address: string;
 }
 
-export const namesByAddressAtom = atomFamily<Param, string[]>((param: Param) =>
```
Comment (fbwoolf) [inline src/store/names.ts:17]: Didn't like this one? Not a good use?

File: src/common/hooks/use-publish-hey.ts (hunk 3 above)
Comment (aulneau) [reply src/common/hooks/use-publish-hey.ts:53]: I'm passing another onFinish from the component to this hook

File: src/store/names.ts (hunk 5 above)
Comment (aulneau) [reply src/store/names.ts:17]: yeah this was not working unfortunately, it was causing issues. Sometimes when there are async atoms that are used by non async atoms, it creates a loop that causes errors, I'll be making an issue in the jotai repo

File: src/common/hooks/use-publish-hey.ts (hunk 3 above)
Comment (fbwoolf) [reply src/common/hooks/use-publish-hey.ts:53]: I see it now in `handleSubmit`, thanks.

File: src/store/names.ts (hunk 5 above)
Comment (fbwoolf) [reply src/store/names.ts:17]: Ok, interesting, good to know. I didn't see the issue so thanks for catching. 👍

File: src/store/names.ts [hunk 6]
```diff
@@ -53,7 +48,10 @@ export const namesAtom = atomFamily((address: string) =>
     }
 
     try {
-      const names = useNamesByAddress(network.coreApiUrl, address);
```
Comment (fbwoolf) [inline src/store/names.ts:56]: I realize now how dumb this was ...haha.

**Pull request merged**